### PR TITLE
WIP Initial Files

### PR DESCRIPTION
**Pull request by lamc8684**
An initial file to read in the handshake message from the GCS

### CODE REVIEW

**Comment by lamc8684** [comment]
Checklist for this part of the project:

- [ ] Create custom message to add to MAVLink

- [ ] Create protocol to send data transmission handshake

- [ ] Create protocol to send images from UAV

- [ ] Create protocol to receive images at GCS

File: mavimage/GPS.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
+from pymavlink import mavutil
+import sys
+from optparse import OptionParser
+from dronekit import connect, VehicleMode
+
+class GPS:
+
+    def __init__(self, connection):
+        parser = OptionParser("readdata.py [options]")
```
Comment (mrshannon) [inline mavimage/GPS.py]: Why is there an option parser inside the GPS class?

File: mavimage/GPS.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        connection_string = args.connect
+        sitl = None
+
+        # Start SITL if no connection string specified
+        if not connection_string:
+            import dronekit_sitl
+            sitl = dronekit_sitl.start_default()
+            connection_string = sitl.connection_string()'''
+        connection_string = self.connection
+        vehicle = connect(connection_string, wait_ready=True)
```
Comment (mrshannon) [inline mavimage/GPS.py]: You cannot make a connection inside of a handler, you must use the MAVLinkConnection given to you in the callback.

File: mavimage/GPS.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+        parser.add_option("--source-system", dest='SOURCE_SYSTEM', type='int', default=255,
+                          help='MAVLink source system')
+        parser.add_option("--showmessages", dest="showmessages", action='store_true',
+                          help="show incoming messages", default=False)
+        (opts, args) = parser.parse_args()
+
+        if opts.device is None:
+            print("Please select a serial device")
+            sys.exit(1)
+        self.connection = mavutil.mavlink_connection(opts.device, baud=opts.baudrate)
```
Comment (mrshannon) [inline mavimage/GPS.py]: I am not 100% sure but I think storing a connection will cause a circular dependency and thus the garbage collector will not be able to free the memory (memory leak).

**Pull request merged**